juju/juju
# feat(machine): support machine cloud instance model migration

This commit adds support for machine cloud instances model migration operations for the machine domain.

As a bonus: machine's method InstanceId is renamed to InstanceID so it adheres to idiomatic golang.





## Checklist




## QA steps

Model migration should work (3.6->4.0 and 4.0->4.0, please test and try to break it), and in the target model the machine cloud instance should be correctly inserted in dqlite:
```
juju bootstrap localhost tgt
juju bootstrap localhost src
juju add-model m
juju deploy ubuntu 
juju migrate src:admin/m tgt
```
At this point we can switch to tgt and check everything is OK:
```
juju switch tgt:admin/m
# make sure you are PWDd at juju root dir
make repl-install
make repl
# retrieve the model db UUID: select * from model
DB_NAME={modeldbUUID} make repl
dqlite> select * from machine_cloud_instance
machine_uuid                            instance_id     arch    mem     root_disk       root_disk_source        cpu_cores       cpu_power       availability_zone_uuid  virt_type
37db5c6b-3872-4d92-8065-96a004219c78    juju-44d1ae-0   amd64   0       0        
```

## Links

**Jira card:** JUJU-6359
